--- a/llvm-interface/gnatllvm-codegen.adb
+++ b/llvm-interface/gnatllvm-codegen.adb
@@ -181,14 +181,6 @@
             CPU        := new String'(Switch_Value (S, "-march="));
          end if;

-      --  -mabi= tells the code generator which ABI to use on some
-      --  platforms
-
-      elsif Starts_With (S, "-mabi=") then
-         To_Free         := ABI;
-         ABI             := new String'(Switch_Value (S, "-mabi="));
-         Tagged_Pointers := ABI.all = "purecap";
-
       --  We support -mXXX and -mno-XXX by adding +XXX or -XXX, respectively,
       --  to the list of features.

@@ -613,13 +605,12 @@
       end;

       Target_Machine    :=
-        Create_Target_Machine_With_ABI
+        Create_Target_Machine
           (T          => LLVM_Target,
            Triple     => Normalized_Target_Triple.all,
            CPU        => CPU.all,
            Features   => Features.all,
-           ABI        => ABI.all,
            Level      => Code_Gen_Level,
            Reloc      => Reloc_Mode,
            Code_Model => Code_Model);
